lib/cloud/gcp/kubernetes.go

```diff
@@ -253,19 +254,44 @@ func convertLocationToGCP(location string) string {
 // getTLSConfig creates a rest.Config for the given cluster with the specified
 // Bearer token for authentication.
 func getTLSConfig(cluster *containerpb.Cluster, tok string) (*rest.Config, error) {
-	if cluster.MasterAuth == nil {
-		return nil, trace.BadParameter("cluster.MasterAuth was not set and is required")
-	}
-	ca, err := base64.StdEncoding.DecodeString(cluster.MasterAuth.ClusterCaCertificate)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
+	dnsConfig := cluster.GetControlPlaneEndpointsConfig().GetDnsEndpointConfig()
+	ipConfig := cluster.GetControlPlaneEndpointsConfig().GetIpEndpointsConfig()
+	switch {
+	case dnsConfig.GetAllowExternalTraffic():
+		if dnsConfig.GetEndpoint() == "" {
+			return nil, trace.BadParameter("dns endpoint was not set and is required")
+		}
 
-	return &rest.Config{
-		Host:        fmt.Sprintf("https://%s", cluster.Endpoint),
-		BearerToken: tok,
-		TLSClientConfig: rest.TLSClientConfig{
-			CAData: ca,
-		},
-	}, nil
+		return &rest.Config{
+			Host:        "https://" + dnsConfig.GetEndpoint(),
+			BearerToken: tok,
+		}, nil
+	case ipConfig.GetEnabled():
```

Comment:
Is it possible for `dnsConfig.GetAllowExternalTraffic` to be true while `dnsConfig.GetEndpoint()` returns no endpoint? In this case, should we try to use `ipConfig` instead of failing?